src/main/java/com/leverx/learningmanagementsystem/course/job/service/CourseNotificationSender.java

```diff
@@ -0,0 +1,27 @@
+package com.leverx.learningmanagementsystem.course.job.service;
+
+import com.leverx.learningmanagementsystem.email.service.impl.EmailService;
+import jakarta.mail.MessagingException;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.mail.MailException;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+@Slf4j
+public class CourseNotificationSender {
+
+    private final EmailService emailService;
+
+    @Async("courseNotificationThreadPoolExecutor")
+    public void tryToSendCourseNotification(String email, String subject, String body) {
+        try {
+            log.info("Current thread: {}", Thread.currentThread().getName());
```

Comment:
it's better to change logger message to something like "Email sending is started" and in logback-spring.xml change message pattern to contain current thread info:
```
<pattern>%date %blue(%-5level) --- [${applicationName}:${applicationVersion}] - %magenta([%thread]) - %cyan(%logger) : %msg%n</pattern>
```